.github/workflows/build_and_test.yml

```diff
@@ -805,3 +853,27 @@ jobs:
       with:
         name: unit-tests-log-docker-integration--8-${{ inputs.hadoop }}-hive2.3
         path: "**/target/unit-tests.log"
+
+  # Note that: there are only GHCR creation permission for secrets.GITHUB_TOKEN
+  # If you want to clean up the CI images, you need to:
+  # - 1. Generate the token from https://github.com/settings/tokens with `write:packages` and `delete:packages`
+  # - 2. Add the the token as secrets `GHCR_DEL` in https://github.com/{username}}/spark/settings/secrets/actions
+  infra-image-post:
+    # Always runs after pyspark/sparkr/lint have completed, regardless of whether they were successful
+    if: always()
+    needs: [pyspark, sparkr, lint]
+    runs-on: ubuntu-20.04
+    env:
+      GHCR_DEL: ${{ secrets.GHCR_DEL }}
```

Comment:
https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow

need to define secrets in main workflow